src/main/scala/workflow/WorkflowUtils.scala

```diff
@@ -0,0 +1,124 @@
+package workflow
+
+import scala.collection.mutable.ArrayBuffer
+
+
+object WorkflowUtils {
+  def instructionsToPipeline[A, B](instructions: Seq[Instruction]): Pipeline[A, B] = {
+    val nodes = new ArrayBuffer[Node]()
+    val dataDeps = new ArrayBuffer[Seq[Int]]()
+    val fitDeps = new ArrayBuffer[Option[Int]]()
```

Comment:
would this be easier to write with a recursive/functional definition rather than one where we build up the state mutably? It doesn't matter becuase all these datastructures are internal, but i'm thinking about how to tighten up/reduce code complexity.